docs/object-utils.md

```diff
@@ -34,4 +34,11 @@ Similar to `groupBy`, but the value is a single element, in case of duplicated v
 
 `convertDateFieldsToIsoString<Input extends object>(object: Input | Input[],): ExactlyLikeWithDateAsString<Input> | ExactlyLikeWithDateAsString<Input>[]`
 
-Transform an input object with `Date` members into an object with the corresponding `Date` members replaced by their ISO string representation. This transformation is done recursively, covering nested objects as well.
\ No newline at end of file
+Transform an input object with `Date` members into an object with the corresponding `Date` members replaced by their ISO string representation. This transformation is done recursively, covering nested objects as well.
+
+`export function deepClone<T>(object: T): T`
+
+Return a deep clone copy of an object. This implementation cloning has these limitations:
+- Cannot clone functions, dates, regex, undefined, infinity, and -0
```

Comment:
I'm concerned about not being able to copy dates, tbh, it's a very common case. everything else is fine when we are talking about objects.

Reply:
Switched to `structuredClone`